### DISCUSSION TITLE
python速速入门

### DISCUSSION THREAD

**Discussion by ZhangAilan** [3S-Academy-NJTech/3S-Learning#1] [Show and tell]
# python速速入门
人生苦短，当你不懂代码不懂数据结构不懂计算机原理却需要通过编程来完成本专业的任务时，python绝对适合你。

尤其当ChatGPT这样的大语言模型出现后，你再也不必花费数十个小时学习那些大部头的视频课最终却迷失在语法的迷宫中，只需提出明确自己的需求，GPT会告诉你答案。就像这样：python中如何定义函数？如何将角度转为弧度？我有10组xy坐标，如何存储并进行累加计算？

我的初次尝试是《数字测图》实习需要计算[闭合导线近似平差以及坐标计算](https://blog.csdn.net/weixin_74671162/article/details/130463185?spm=1001.2014.3001.5501)，逻辑很简单，当我无法实现某一步的计算就去询问AI，如果你不知道从何开始，不妨点击链接重写一遍，可以要求按行解释代码来学习语法。类似的案例还有[三角高程计算](https://blog.csdn.net/weixin_74671162/article/details/131257509?spm=1001.2014.3001.5501)、[Tkinter绘制导线缩略图](https://zhuanlan.zhihu.com/p/673805012)

python像瑞士军刀一样，它封装好了各种功能，这依赖于其开源的生态和无比强大的拓展库，这个[视频](https://www.bilibili.com/video/BV11z421m7wF/?spm_id_from=333.337.search-card.all.click&vd_source=012f01c73f97da5b9202e8ac4e57e620)介绍了很多经典的库以及python的历史，你只需关注程序的逻辑和调用接口，然后pip install xxx，在.py中import xxx即可。

哦对了，安装python看[这里](https://liaoxuefeng.com/books/python/install/)。

## 虚拟环境
最折磨人的莫过于当你使用了不同版本的python包和库时引起了相互冲突，而虚拟环境可以为每个项目提供独立的python解释器和包管理环境，从而避免冲突。简言之，**一个项目一个虚拟环境！**

在此之前，你最好安装一个集成开发环境（IDE），可以提供语法高亮、代码补全以及调试等功能（你要是[Vim](https://www.runoob.com/linux/linux-vim.html)大佬我也没话说～）。这里推荐VS Code，它本质上是一个文本编辑器，但里面内置了很多强大的插件来帮助你（看[这里](https://blog.csdn.net/qq_30640671/article/details/109704940)安装）。当然pycharm也可以，不过我每次关闭项目都要加载好久，不喜欢～

言归正传，在此提出两种虚拟环境创建方式：venv和conda

### venv
Python 提供了内置模块 venv 来创建虚拟环境

创建虚拟环境：`python -m venv myenv `  myenv为虚拟环境名称

激活虚拟环境：
windows `myenv\Scripts\activate`
linux `source myenv/bin/activate`

安装依赖包（以numpy举例）：`pip install numpy`   
若安装速度过慢，可尝试[换源](https://blog.csdn.net/xuezhangjun0121/article/details/81664260)

若删除虚拟环境，删除对应myenv文件夹即可

贴一个全方位[教程](https://blog.csdn.net/blxmove/article/details/137072545)
### conda
安装及创建过程可见[此](https://blog.csdn.net/zhouzhiwengang/article/details/129952728)

---
到此就可以开始code了，之后想到了再补充～

**Comment by sssshzzi** [comment]
好

**Comment by citynew** [comment]
好极了

**Comment by quater4** [comment]
豪
